README.md

```diff
@@ -86,23 +83,30 @@ constructor(ctx: any, env: Env, options?: {
 #### Methods
 
 ##### Lifecycle Methods
+All lifecycle methods can be implemented as async if needed.
 
 - `onStart()`: Called when container starts successfully - override to add custom behavior
 - `onStop()`: Called when container shuts down - override to add custom behavior
 - `onError(error)`: Called when container encounters an error - override to add custom behavior
+- `onActivityExpired()`: Called when the activity is expired - override to add custom behavior, like communicating with the container to see if it should be shutdown.
+
+By default, it calls `ctx.container.destroy()`.
+If you don't stop the container here, it will just renew the activity. It will be called again once it expires.
 
 ##### Container Methods
 
 - `fetch(request)`: Default handler to forward HTTP requests to the container. Can be overridden.
-- `containerFetch(...)`: Sends an HTTP or WebSocket request to the container. Supports both standard fetch API signatures:
+- `containerFetch(...)`: Sends an HTTP request to the container. Supports both standard fetch API signatures:
   - `containerFetch(request, port?)`: Traditional signature with Request object
   - `containerFetch(url, init?, port?)`: Standard fetch-like signature with URL string/object and RequestInit options
-  Either port parameter or defaultPort must be specified. Automatically detects WebSocket upgrade requests.
+  Either port parameter or defaultPort must be specified.
+  When you call any of the fetch functions, the activity will be automatically renewed, and if the container will be started if not already running.
 - `start()`: Starts the container if it's not running and sets up monitoring, without waiting for any ports to be ready.
 - `startAndWaitForPorts(ports?, maxTries?)`: Starts the container using `start()` and then waits for specified ports to be ready. If no ports are specified, uses `requiredPorts` or `defaultPort`. If no ports can be determined, just starts the container without port checks.
-- `stop(reason?)`: Stops the container
+- `stop(signal = SIGTERM)`: Sends the specified signal to the container.
```

Comment:
We should add a changelog section to the readme so we can call out interface changes like this

Reply:
also I dont see why we should provide a reason to stop, we basically don't use it at all for now